.github/workflows/labeler.yml

```diff
@@ -0,0 +1,64 @@
+name: Labeler
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+    paths:
+      - '../labels.yml'
+
+
+jobs:
+
+  label-syncer:
+    name: Syncer
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: micnncim/action-label-syncer@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  labeler:
+    name: Set labels
+    needs: [label-syncer]
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+
+    # Label based on modified files
+    - name: Label based on changed files
+      uses: actions/labeler@v4
+      with:
+        repo-token: "${{ secrets.GITHUB_TOKEN }}"
+        sync-labels: ''
+
+    # Label based on branch name
+    - uses: actions-ecosystem/action-add-labels@v1
+      if: |
+        startsWith(github.event.pull_request.head.ref, 'doc') || 
+        startsWith(github.event.pull_request.head.ref, 'docs')
+      with:
+        labels: documentation
+
+    - uses: actions-ecosystem/action-add-labels@v1
+      if: |
+        startsWith(github.event.pull_request.head.ref, 'maint') ||
+        startsWith(github.event.pull_request.head.ref, 'no-ci') ||
+        startsWith(github.event.pull_request.head.ref, 'docker') ||
+        startsWith(github.event.pull_request.head.ref, 'ci')
+      with:
+        labels: maintenance
+
+    - uses: actions-ecosystem/action-add-labels@v1
+      if: startsWith(github.event.pull_request.head.ref, 'feat')
+      with:
+        labels: |
+          enhancement
+    - uses: actions-ecosystem/action-add-labels@v1
+      if: |
+        startsWith(github.event.pull_request.head.ref, 'fix') ||
+        startsWith(github.event.pull_request.head.ref, 'patch')
+      with:
+        labels: bug
```

Comment:
All this is great as long as people name their branches like:

* feat/**
* doc(s)/**
* maint/**
* fix/**
* ...
We should clarify to people that branches are not named like:

* myname/**

I've already seen this in other repos. Maybe you could talk about this in tomorrow's meeting @jorgepiloto?We already have many repos with the autolabeler. And more to come! =)

Reply:
@jorgepiloto @RobPasMue 
We have that documented here in the "Contributing" section.
https://dev.docs.pyansys.com/how-to/contributing.html?highlight=branch#creating-a-branch

However, I agree we should remind people to follow it as much as possible. I will also remind this best practice for other repos.